packages/driver/src/cypress/proxy-logging.ts

```diff
@@ -343,6 +337,29 @@ export default class ProxyLogging {
     proxyRequest.log?.end()
   }
 
+  private updateRequestWithResponse (responseReceived: BrowserResponseReceived): void {
+    const proxyRequest = _.find(this.proxyRequests, ({ preRequest }) => preRequest.requestId === responseReceived.requestId)
+
+    if (!proxyRequest) {
+      return debug('unmatched responseReceived event %o', responseReceived)
+    }
+
+    if (proxyRequest.xhr && proxyRequest.xhr.xhr.readyState !== 4) {
+      const oldOnload = proxyRequest.xhr.xhr.onload
+      const proxyLogging = this
+
+      proxyRequest.xhr.xhr.onload = function (...args) {
+        if (oldOnload) {
+          oldOnload.call(this, ...args)
+        }
+
+        proxyLogging.updateProxyRequestWithResponse(proxyRequest, responseReceived)
+      }
+    }
+
+    this.updateProxyRequestWithResponse(proxyRequest, responseReceived)
```

Comment:
Should this be in an `else` or do we need `updateProxyRequestWithResponse` called twice?

Reply:
Doh. That was an artifact of refactoring. Fixed here: [`70b07df` (#21373)](https://github.com/cypress-io/cypress/pull/21373/commits/70b07dfe4acb71861a97c01660c3017e82aae5b1)